receiver/httpcheckreceiver/scraper.go

```diff
@@ -85,6 +87,26 @@ func (h *httpcheckScraper) scrape(ctx context.Context) (pmetric.Metrics, error)
 					h.mb.RecordHttpcheckStatusDataPoint(now, int64(0), h.cfg.Targets[targetIndex].Endpoint, int64(statusCode), req.Method, class)
 				}
 			}
+
+			matchContent := h.cfg.Targets[targetIndex].MatchContent
+			if err == nil && len(matchContent) > 0 {
+				if body, err := io.ReadAll(resp.Body); err != nil {
+					h.settings.Logger.Error("failed to read response", zap.Error(err))
+				} else {
+					defer resp.Body.Close()
+
+					if len(body) > 0 {
+						if strings.Contains(string(body), matchContent) {
+							h.mb.RecordHttpcheckMatchContentDataPoint(now, int64(1), h.cfg.Targets[targetIndex].Endpoint, matchContent)
+							h.mb.RecordHttpcheckReverseMatchContentDataPoint(now, int64(0), h.cfg.Targets[targetIndex].Endpoint, matchContent)
+						} else {
+							h.mb.RecordHttpcheckMatchContentDataPoint(now, int64(0), h.cfg.Targets[targetIndex].Endpoint, matchContent)
```

Comment:
I'm not sure why you'd need both a metric for content matched and a metric for content not matched, given that you're able to capture that with the 0 and 1 values. Can you simplify to just one metric to match content.